### DISCUSSION TITLE
If any implicit problem with changing the proxy like this?

### DISCUSSION THREAD

**Discussion by doveppp** [encode/httpx#2426] [General] [answered]
change proxy like this:
![image](https://user-images.githubusercontent.com/54977106/197338813-71604410-80a3-4b11-9208-52eae060dc91.png)

because there has not any method to change proxy on the session inplace

**Comment by lovelydinosaur** [comment, marked as answer]
You're using private API there, so it might break on updates to the software.

I'd suggest using a new client instance if you want to change the proxy configuration.
